tests/conftest.py

```diff
@@ -149,103 +179,228 @@ def test_something(remote: Remote):
     ```
     """
 
-    slurm_cluster_hostname = request.param
-
-    if not slurm_cluster_hostname:
+    cluster_name = request.param
+    if not cluster_name:
         pytest.skip("Requires ssh access to a SLURM cluster.")
 
-    # TODO: Re-enable this, but only on tests that say that they run jobs on the
-    # cluster.
-    # with cancel_all_milatools_jobs_before_and_after_tests(slurm_cluster_hostname):
-    return slurm_cluster_hostname
+    clusters_in_maintenance = os.environ.get("CLUSTER_DOWN", "").split(",")
+    if cluster_name in clusters_in_maintenance:
+        pytest.skip(reason=f"Cluster {cluster_name} is down for maintenance.")
+        # TODO: Seems not possible to add this marker to all tests?.
+        # request.node.add_marker(
+        #     pytest.mark.xfail(
+        #         reason=f"Cluster {cluster_name} is down for maintenance.",
+        # raises=subprocess.CalledProcessError,
+        #     )
+        # )
+    return cluster_name
+
+
+@pytest.fixture(scope="session")
+def job_name(request: pytest.FixtureRequest) -> str | None:
+    # TODO: Make the job name different based on the runner that is launching tests, so
+    # that the `launches_job_fixture` doesn't scancel the test jobs launched from
+    # another runner (e.g. me on my dev machine or laptop) on a cluster
+
+    return getattr(request, "param", get_job_name_for_tests(request))
+
+
+def get_job_name_for_tests(request: pytest.FixtureRequest) -> str | None:
+    this_machine = socket.gethostname()
+    this_test_name = request.node.name
+    job_name = f"{JOB_NAME}_{this_test_name}_{this_machine}"
+    if in_self_hosted_github_CI:
+        # NOTE: We use this in the `build.yml` file to limit concurrent jobs for the
+        # same branch/workflow
+        # group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+        # here we do something similar
+        github_ref = os.environ["GITHUB_REF"]
+        workflow_name = os.environ["GITHUB_WORKFLOW"]
+        job_name += f"_{workflow_name}_{github_ref}"
```

Comment:
Sorry, I don't quite understand. This is trying to make a unique job name so the self-hosted runner's jobs on the cluster don't get cancelled or interfered with when running the unit tests on a dev machine. How would you use an environment variable here? Do you mean instead of the hostname of the dev machine, we could set an environment variable?

Reply:
I don't really like to have to know where we are running the tests (github CI for exemple). In general, I think it's easier to understand the behaviour of the code by having a standard interfaces like a set of env vars that are explicitly set before executing the tests instead of having to store in your working memory the different places where things could be affected by the environment. The idea is to have the same behaviour as much as possible, reduce the number of `if`s where we can and have the code as streamed lined as possible. But in the case of this example it doesn't change much